tflite2onnx/op/resize.py

```diff
@@ -0,0 +1,128 @@
+        # half_pixel, pytorch_half_pixel, align_corners, asymmetric, tf_crop_and_resize
+        self.attrs['coordinate_transformation_mode'] = 'half_pixel'
+        # This attribute is valid only if "mode" is "cubic".
+        # The coefficient 'a' used in cubic interpolation.
+        # Two common choice are -0.5 (in some cases of TensorFlow) and -0.75 (in PyTorch).
+        self.attrs['cubic_coeff_a'] = -0.75
+        self.attrs['exclude_outside'] = 0
+        self.attrs['extrapolation_value'] = 0.0
+        # Three interpolation modes: nearest (default), linear and cubic.
+        # The "linear" mode includes linear interpolation for 1D tensor
+        # and N-linear interpolation for N-D tensor
+        # (for example, bilinear interpolation for 2D tensor).
+        # The "cubic" mode includes cubic interpolation for 1D tensor
+        # and N-cubic interpolation for N-D tensor
+        # (for example, bicubic interpolation for 2D tensor).
+        self.attrs['mode'] = 'nearest'
+        # Four modes: round_prefer_floor (default, as known as round half down),
+        # round_prefer_ceil (as known as round half up), floor, ceil.
+        # Only used by nearest interpolation.
+        # It indicates how to get "nearest" pixel in input tensor from x_original,
+        # so this attribute is valid only if "mode" is "nearest".
+        self.attrs['nearest_mode'] = 'round_prefer_floor'
+
+        self.setInited()
+
+    @property
+    def type(self):
+        return 'Resize'
+
+    @property
+    def isRESIZE_BILINEAR(self):
+        op = self.tflite
+        opcode = self.model.OperatorCodes(op.OpcodeIndex()).BuiltinCode()
+        return opcode is tflite.BuiltinOperator.RESIZE_BILINEAR
+
+    @property
+    def isRESIZE_NEAREST_NEIGHBOR(self):
+        op = self.tflite
+        opcode = self.model.OperatorCodes(op.OpcodeIndex()).BuiltinCode()
+        return opcode is tflite.BuiltinOperator.RESIZE_NEAREST_NEIGHBOR
+
+    def parse(self):
+        logger.debug("Parsing %s...", self.type)
+        op = self.tflite
+        opcode = self.model.OperatorCodes(op.OpcodeIndex()).BuiltinCode()
+        assert(opcode is tflite.BuiltinOperator.RESIZE_BILINEAR
+               or tflite.BuiltinOperator.RESIZE_NEAREST_NEIGHBOR)
+
+        assert(op.InputsLength() >= 2), "Only first two arguments: image and size, are compulsory"
+        assert(op.OutputsLength() == 1)
+
+        # image
+        ilayout = Layout('NHWC', 'NCHW')
+        im = self.parseInput(0, ilayout)
+
+        # Fill 'ROI' empty temporarily
+        # because 'tf_crop_and_resize' was not found in BuiltinOptions
+        # of ResizeBilinear and ResizeNearestNeighbor
+        holder = self.TFactory.createVector(im, np.array([]))
+        holder.parse()
+        holder.addConsumer(self)
+        self.inputs.append(holder)
+
+        # Fill empty to scales temporarily
+        sc = self.TFactory.createVector(im, np.array([]))
+        sc.parse()
+        self.inputs.append(sc)
```

Comment:
Do we need to add `Resize` as a consumer of `sc`?

Reply:
Sorry to jump in, but I happened to be in need of Resize myself when I found this.

Adding self as a consumer of sc fails because "sc" here returned is the same tensor object from previous line, i.e. "holder". This happens because both of their names resolve to the same. I'm guessing they need to be forced to be unique somehow?

Reply:
That makes sense, we may use `holder` directly currently. And, welcome @ashikns :)